phconvert/pqreader.py

```diff
@@ -643,13 +822,29 @@ def _correct_overflow_nsync(timestamps, detectors, overflow_ch, overflow):
 
 def _correct_overflow_nsync_naive(timestamps, detectors, overflow_ch, overflow):
     """Slow implementation of `_correct_overflow_nsync` used for testing.
-    """
+    """ 
     overflow_correction = 0
     for i in range(detectors.size):
         if detectors[i] == overflow_ch:
             overflow_correction += (overflow * timestamps[i])
         timestamps[i] += overflow_correction
-
+def _t3r_correct_overflow(timestamps, valid, overflow_ch, overflow):
+    """Apply overflow correction when each overflow has a special timestamp.
+    """
+    overflow_correction = 0;
+    for i in xrange(valid.size):
+        if valid[i] == 0:
+            overflow_correction += overflow
+        timestamps[i] += overflow_correction
```

Comment:
This function appears identical to `_correct_overflow1` with detectors renamed to valid. If this is the case please remove and use  `_correct_overflow1`  instead.

Reply:
Indeed. It works with _correct_overflow1 by setting the overflow_ch=0. The extra functions is removed.